File: 2021/April/Day 17/set-mismatch.py
# source : https://leetcode.com/problems/set-mismatch/submissions/


class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """

        missing = 0;
        visited = [];
        repeated =0;
        for i in nums:
            if(i in visited):
                repeated = i;
                break
            else:
                visited.append(i)

        for i in range(1,len(nums)+1):
            if(i not in nums):
                missing = i;
                break
        return [repeated,missing]

File: 2022/Oktober/Day 23/setMismatch.cpp
// Source : https://leetcode.com/problems/set-mismatch/submissions/
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int val= NULL;
        int missing=NULL;
        for(int i=1;i<=size(nums);i++)
        {

            if(missing==NULL && std::find(nums.begin(),nums.end(),i)==nums.end())
            {
               missing=i;

            }

            if(val==NULL && std::count(nums.begin(),nums.end(),nums[i-1])>1)
            {
               val=nums[i-1];
            }
        }
        return {val,missing};
    }
};
